packages/next/src/build/webpack/loaders/next-flight-loader/module-proxy.ts

```diff
@@ -92,11 +211,22 @@ const proxyHandlers: ProxyHandler<object> = {
 }
 
 export function createProxy(moduleId: string) {
-  const moduleReference = {
-    $$typeof: MODULE_REFERENCE,
-    filepath: moduleId,
-    name: '*', // Represents the whole object instead of a particular import.
-    async: false,
-  }
-  return new Proxy(moduleReference, proxyHandlers)
+  const clientReference = Object.defineProperties(
+    function () {
```

Comment:
This is the issue. Because Webpack feature tests if this is already a module or if it should make a fake module around it, it checks if the module is `typeof ... === "object"`.

Therefore we can't make this one a function. We have to go back to the previous object form. This is unfortunate because we can't warn if someone does:

```
const fn = require('fn');
fn();
```

Maybe we assume they probably use an ESM compatibility layer.